app/controllers/stories_controller.rb

```diff
@@ -36,31 +36,29 @@ def show
   private
 
   def redirect_to_changed_username_profile
-    if @user = User.find_by_old_username(params[:username].tr("@", "").downcase)
+    potential_username = params[:username].tr("@", "").downcase
+    user_or_org = User.find_by("old_username = ? OR old_old_username = ?", potential_username, potential_username) ||
+      Organization.find_by("old_slug = ? OR old_old_slug = ?", potential_username, potential_username)
+    if user_or_org.class.name == "User"
+      @user = user_or_org
       redirect_to @user.path
       return
-    end
-    if @user = User.find_by_old_old_username(params[:username].tr("@", "").downcase)
-      redirect_to @user.path
+    elsif user_or_org.class.name == "Organization"
+      @organization = user_or_org
+      redirect_to @organization.path
       return
+    else
+      not_found
     end
-    not_found
   end
 
   def handle_possible_redirect
-    if @user = User.find_by_old_username(params[:username].tr("@", "").downcase)
-      if @user.articles.find_by_slug(params[:slug])
-        redirect_to "/#{@user.username}/#{params[:slug]}"
-        return
-      end
-    end
-    if @user = User.find_by_old_old_username(params[:username].tr("@", "").downcase)
-      if @user.articles.find_by_slug(params[:slug])
-        redirect_to "/#{@user.username}/#{params[:slug]}"
-        return
-      end
-    end
-    if @organization = @article.organization
+    potential_username = params[:username].tr("@", "").downcase
+    @user = User.find_by("old_username = ? OR old_old_username = ?", potential_username, potential_username)
+    if @user&.articles&.find_by_slug(params[:slug])
+      redirect_to "/#{@user.username}/#{params[:slug]}"
+      return
+    elsif @organization = @article.organization
```

Comment:
This keeps the logic consistent with existing behavior, where it redirects to an organization's article if `params[:slug]` found an article. This also means we don't have to look for an organization's `old_slug` or `old_old_slug`.